src/main/osd/osd_elements.c

```diff
@@ -783,20 +783,25 @@ static void osdBackgroundCameraFrame(osdElementParms_t *element)
     element->drawElement = false;  // element already drawn
 }
 
+static void toUpperCase(char* dest, const char* src, unsigned maxSrcLength)
+{
+    unsigned i;
+    for (i = 0; i < maxSrcLength; i++) {
+        if (src[i]) {
+            dest[i] = toupper((unsigned char)src[i]);
+        } else {
+            break;
+        }
+    }
+    dest[i] = '\0';
+}
+
```

Comment:
i like your idea of getting rid of the IF statement.
However the changes around MAX_NAME_LENGTH will require more refactoring. For example currently the string fields in the structures are accomodating for the extra char like that:
```
 typedef struct pilotConfig_s {
     char name[MAX_NAME_LENGTH + 1];
     char displayName[MAX_NAME_LENGTH + 1];
 } pilotConfig_t;

```

Reply:
`MAX_NAME_LENGTH` is the length of chars you can input. The +1 at the end is for the necessary termination char that signals the end of the array and it is **always necessary**. Therefore I don't think we need any refactoring at all in `pilotConfig_t`.

> It can be better to call it like:
> 
> toUpperCase(element->buff, pilotConfig()->name, MAX_NAME_LENGTH+1);

I don't think so, because then you have to think about not forgetting to increment the max length by +1 to accomodate for the termination char that is always necessary. We should not do this and let users input MAX_NAME_LENGTH explicitly into the function and let the function handle the array termination automatically.

Reply:
> @limonspb `MAX_NAME_LENGTH` is the length of chars you can input. The +1 at the end is for the necessary null-terminated char that signals the end of the array and it is **always necessary**. Therefore I don't think we need any refactoring at all in `pilotConfig_t`.

Agree.

> I don't think so, because then you MUST NOT forget to always increment maxSrcLength by +1 to accomodate for the termination char that is always necessary.
If I forget the worst case that could happen is that the string will be 1 character shorter - nothing terrible and probably we will be able to find it out. On the other hand current proposal writes `MAX_NAME_LENGTH + 1` characters to destination but parameter `maxSrcLength` is `MAX_NAME_LENGTH` which seems to be confusing for me and we can end up with overwriting one more character in the destination which could be harmful.

Reply:
@KarateBrot  cmon i know what is null-terminated string :))

@daleckystepan i think in your example it should **not** be `ARRAYLEN(pilotConfig()->name)` instead it should be `strlen(pilotConfig()->name)`. And `strlen()` returns you the number of bytes without `\0`. So I agree with @KarateBrot here that we should hide the +1 logic inside the function.

Reply:
i would say that the current implementation uses `MAX_NAME_LENGTH` similar to `strlen` because both of them do not include `\0`. MAX_NAME_LENGTH == strlen exactly.
But you are right that we can just omit `MAX_NAME_LENGTH`. Similar to how it is omitted in `strcpy`. I think someone in the past just added the `MAX_NAME_LENGTH ` check in the loop for safety, in case somehow by mistake the `pilotConfig()->name` will be filled with real chars including the last hidden symbol. So this `i < maxSrcLength` is there just to avoid "infinite" loop and corrupting everything in case somehow someone messes up the string.

Reply:
Agree. So we have discussed it very carefully :) Your PR so you can decide what to choose.

Can you please at least accept accept my idea of moving condition to for loop? Getting rid off if, else, break. Thank you.

Reply:
@daleckystepan absolutely :) Done

Reply:
deleted the branch accidentally, facepalm

Reply:
recovered )

Reply:
Misunderstanding. I didn't want to sound like i an explaining a null terminator to you, lol.